Design-concepts.md

```diff
@@ -260,6 +264,299 @@ For FuseML to be functional, the following design elements and components are ne
+
+The two segments of the workflow are detailed separately in the next sections and they both involve running the fuseml CLI. As a preparation step, the user must configure how the CLI can contact the FuseML API. Several mechanisms could be used for this, all of which should be supported:
+* environment variable (e.g. `export FUSEML_URL=http://fuseml.10.20.30.nip.io`)
+* command line parameter (e.g. `fuseml --url http://fuseml.10.20.30.nip.io`)
+* configuration file (e.g. `~/.fuseml/config`)
+
+Furthermore, when authentication and authorization features are implmented, user credentials will also need to be provided in a similar fashion.
+
+#### Publishing Codesets
+
+In this part of the workflow, the actor responsible for writing MLFlow code (e.g. data scientist) makes the code available to FuseML in the form of a codeset artifact. Several ways of registering FuseML codesets can be imagined, but the simplest one is by running the fuseml CLI on the same machine where the code is located:
+
+* first, create a project - a concept used to group together several codesets and possible other objects, including pipeline templates, with the purpose of applying other operations to the group as a whole (not yet detailed in this document). On the codeset side of things, this is represented by a git organization used as a parent for all codesets belonging to the project. This operation should be restricted so that only FuseML administrators or delegated roles can execute it, but for the purpose of this exercise, we'll assume everyone can do it: 
+
+  ```
+  fuseml create project --name my-ml-project --description "My first ML project with FuseML"
+  ```
+
+  What happens under the hood:
+    * the fuseml CLI calls the remote FuseML REST API to create a project, e.g.;
+
+      POST on http://fuseml.10.20.30.nip.io/api/project with body:
+      ```
+      project:
+        name: my-ml-project
+        description: |
+          My first ML project with FuseML
+      ```
+
+    * the FuseML core service calls the gitea API to create a `my-ml-project` organization (if not yet present)
+    * if everything works out ok, a success code is returned to the CLI
+
+* next, publish the code in the form of a codeset artifact, indicating the local folder where the code is located, as well as the parent project and the type of codeset (by specifying labels):
+
+  ```
+  fuseml push codeset --name my-mlflow-app --description "My first MLFlow application with FuseML" --location local/path/to/code --label mlflow --project my-ml-project --branch main
```

Comment:
`fuseml codeset push`

Reply:
will fix in the next iteration, here and everywhere else, to have the subject token first (project, codeset etc) and the verbs second (push, create etc)